public/js/application.js

```diff
@@ -0,0 +1,38 @@
+$(document).ready(function() {
+  // This is called after the document has loaded in its entirety
+  // This guarantees that any elements we bind to will exist on the page
+  // when we try to bind to them
+
+  // See: http://docs.jquery.com/Tutorials:Introducing_$(document).ready()
+  $('#login_form').hide(function(){
```

Comment:
setting the click handler to show the login_form in a hide callback is a little weird. You should make the login form hidden by default with CSS or inline styles in your html. Then just assign the click callback on its own.